### PR TITLE
Refs #35361 - Add development docs

### PR DESCRIPTION
**Pull request by ShimShtein**

### CODE REVIEW

File: developer_docs/how_to_create_a_plugin.asciidoc [hunk 1]
```diff
@@ -465,6 +465,36 @@ Possible mount points:
...
+* `class` - additional html classes to put on the `<th>` element
+* `attr_callbacks` - a hash where the key is the name of html attribute, and the value is a function in the form `->(host) { attribute_value }`
+* `callback` - a function that receives the host model and returns html content for the `<th>` element: `->(host) { "<span>...</span>".html_safe } `
+
+`hosts_table_column_content` supports the following attributes:
+* `key` - the name that will be used to choose the column
+* `width` - width percentage of the column
+* `class` - additional html classes to put on the `<th>` element
+* `attr_callbacks` - a hash where the key is the name of html attribute, and the value is a function in the form `->(host) { attribute_value }`
+* `callback` - a function that receives the host model and returns html content for the `<th>` element: `->(host) { "<span>...</span>".html_safe } `
```
Comment (sbernhard) [inline developer_docs/how_to_create_a_plugin.asciidoc]: Would be great to have a example how to use it. Or link to a page which already uses these pagelet.

Comment (ShimShtein) [reply developer_docs/how_to_create_a_plugin.asciidoc]: @sbernhard Added a reference to the core repo that uses these extension points

File: developer_docs/how_to_create_a_plugin.asciidoc [hunk 2]
```diff
@@ -465,6 +465,36 @@ Possible mount points:
...
+
+`hosts_table_column_header` supports the following attributes:
+
+* `key` - the name that will be used to choose the column
+* `label` - a string that will be shown in the header row
+* `sortable` - true/false value that indicates whether the column should support sorting
+* `width` - width percentage of the column
+* `class` - additional html classes to put on the `<th>` element
+* `attr_callbacks` - a hash where the key is the name of html attribute, and the value is a function in the form `->(host) { attribute_value }`
+* `callback` - a function that receives the host model and returns html content for the `<th>` element: `->(host) { "<span>...</span>".html_safe } `
```
Comment (pkoprda) [inline developer_docs/how_to_create_a_plugin.asciidoc]:
```suggestion
* `callback` - a function that receives the host model and returns html content for the `<th>` element: `->(host) { "<span>...</span>".html_safe }`
```

File: developer_docs/how_to_create_a_plugin.asciidoc [hunk 3]
```diff
@@ -465,6 +465,36 @@ Possible mount points:
...
+
+`hosts_table_column_content` supports the following attributes:
```
Comment (pkoprda) [inline developer_docs/how_to_create_a_plugin.asciidoc:489]: Add new line for better formatting:
```suggestion
`hosts_table_column_content` supports the following attributes:

```

File: developer_docs/how_to_create_a_plugin.asciidoc (hunk 1 above)
Comment (pkoprda) [inline developer_docs/how_to_create_a_plugin.asciidoc]:
```suggestion
* `callback` - a function that receives the host model and returns html content for the `<th>` element: `->(host) { "<span>...</span>".html_safe }`
```

File: developer_docs/how_to_create_a_plugin.asciidoc [hunk 4]
```diff
@@ -465,6 +465,36 @@ Possible mount points:
...
+* `key` - the name that will be used to choose the column
+* `width` - width percentage of the column
+* `class` - additional html classes to put on the `<th>` element
+* `attr_callbacks` - a hash where the key is the name of html attribute, and the value is a function in the form `->(host) { attribute_value }`
+* `callback` - a function that receives the host model and returns html content for the `<th>` element: `->(host) { "<span>...</span>".html_safe } `
+
+You can find usage of those pagelets in the [core repository](https://github.com/theforeman/foreman/blob/fd2d3f160ca7db0d56cac1e61ce6a474b5cf7def/config/initializers/foreman_register.rb#L8)
```
Comment (pkoprda) [inline developer_docs/how_to_create_a_plugin.asciidoc]:
```suggestion
You can find usage of those pagelets in the https://github.com/theforeman/foreman/blob/fd2d3f160ca7db0d56cac1e61ce6a474b5cf7def/config/initializers/foreman_register.rb#L8[core repository].
```

**Review by pkoprda — CHANGES_REQUESTED** [review]
Thanks @ShimShtein! Maybe it would be great to have some information about `use_pagelet` and how to use it. We use it already in `Content` category: https://github.com/Katello/katello/blob/b7f84e5a6daa3168ea2009d26b17634cd696f0b1/lib/katello/plugin.rb#L272. 
Also I added a few suggestions for better formatting :)

**Comment by ShimShtein** [comment]
@pkoprda fixed all the minor issues.

> Maybe it would be great to have some information about use_pagelet and how to use it.

I think it would require a new paragraph about profiles. Let's leave it for another PR.

**Review by pkoprda — APPROVED** [review]
Okay, I agree, let's leave it for another PR.

Apart from that, LGTM!

**Review by MariaAga — APPROVED** [review]
Thanks everyone for the reviews!

**Pull request merged**